src/components/List/List.tsx

```diff
@@ -286,56 +250,71 @@ export class List extends React.Component {
             },
         );
     };
-    onFilterUpdate = (value) => {
+
+    private onFilterUpdate = (value: string) => {
         if (this.props.onFilterUpdate) {
             this.props.onFilterUpdate(value);
         } else {
             this.onUpdateFilterInternal(value);
         }
     };
-    onItemsRendered = ({visibleStartIndex, visibleStopIndex}) => {
+
+    private onItemsRendered = ({
+        visibleStartIndex,
+        visibleStopIndex,
+    }: {
+        visibleStartIndex: number;
+        visibleStopIndex: number;
+    }) => {
         this.setState({
-            visibleStartIndex,
-            visibleStopIndex,
             pageSize: visibleStopIndex - visibleStartIndex,
         });
     };
-    onItemMouseMove = (index) => {
+
+    private onItemMouseMove = (index: number) => {
         if (!this.state.sorting) {
             this.activateItem(index, false);
         }
     };
-    onMouseLeave = () => {
+
+    private onMouseLeave = () => {
         this.deactivate();
     };
-    onSortStart = () => {
+
+    private onSortStart = () => {
         this.setState({sorting: true});
     };
-    onSortEnd = (param) => {
+
+    private onSortEnd = (params: ListSortParams) => {
         if (this.props.onSortEnd) {
-            this.props.onSortEnd(param);
+            this.props.onSortEnd(params);
         }
+
         this.setState({
             sorting: false,
-            activeItem: param.newIndex,
+            activeItem: params.newIndex,
         });
     };
-    getItemHeight = (index) => {
+
+    private getItemHeight = (index: number) => {
         const {itemHeight, virtualized} = this.props;
+
         if (typeof itemHeight === 'function') {
             const {items} = this.state;
             return itemHeight(items[index]);
         } else {
             return virtualized ? Number(itemHeight) || 28 : itemHeight;
         }
     };
-    renderItem = ({index, style}) => {
+
+    private renderItem = ({index, style}: {index: number; style?: React.CSSProperties}) => {
```

Comment:
Let's put inner render methods right after the main render for convenience

Reply:
Ok. I`ll add all of these render methods after public methods

Reply:
and render method should remain the last of public methods